TM1py/Services/RestService.py

```diff
@@ -217,21 +227,38 @@ def __init__(self, **kwargs):
 
         self._sandboxing_disabled = None
 
-        # manage connection pool
-        if "connection_pool_size" in kwargs:
-            self._manage_http_connection_pool(kwargs.get("connection_pool_size"))
+        if self._tcp_keepalive or self._connection_pool_size is not None:
+            self._manage_http_adapter()
 
         # Logging
         if 'logging' in kwargs:
             if self.translate_to_boolean(value=kwargs['logging']):
                 http_client.HTTPConnection.debuglevel = 1
 
-    def _manage_http_connection_pool(self, connection_pool_size: Union[str, int]):
-        self._s.mount(
-            self._base_url,
-            HTTPAdapter(
-                pool_connections=int(connection_pool_size),
-                pool_maxsize=int(connection_pool_size)))
+    def _manage_http_adapter(self):
+        if self._tcp_keepalive:
+            # Set the following socket options
+            # SO_KEEPALIVE: set 1 to enable TCP keepalive
+            # TCP_KEEPIDLE: Time in seconds until the first keepalive is sent
+            # TCP_KEEPINTVL: How often should the keepalive packet be sent
+            # TCP_KEEPCNT: The max number of keepalive packets to send
+            socket_options = urllib3.connection.HTTPConnection.default_socket_options + [(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1),
+                                                                                         (socket.IPPROTO_TCP, socket.TCP_KEEPIDLE, 30),
+                                                                                         (socket.IPPROTO_TCP, socket.TCP_KEEPINTVL, 15),
+                                                                                         (socket.IPPROTO_TCP, socket.TCP_KEEPCNT, 60)]
```

Comment:
Nice work @macsir. Thank you.

One suggestion on the code. Perhaps we could store the default values for `TCP_KEEPIDLE` etc. in a class variable similar to the `HEADERS`.
So that in the rare case that someone wants to change it, they wouldn't need to change the code in the RESTService. 

Kinda like here:
https://github.com/cubewise-code/tm1py/discussions/797#discussioncomment-3533883

Reply:
Hi, @MariusWirtz That's a great idea and will implement it soon.